Extensions/ShareTo/InitialViewController.swift

```diff
@@ -85,11 +88,25 @@ class InitialViewController: UIViewController {
         UIDevice.current.beginGeneratingDeviceOrientationNotifications()
         super.viewDidLoad()
         view.backgroundColor = UIColor(white: 0.0, alpha: UX.alphaForFullscreenOverlay)
-        // This is the view controller for the popup dialog
-        shareViewController = ShareViewController()
-        shareViewController.delegate = self
-        embedController = EmbeddedNavController(parent: self, rootViewController: shareViewController)
         view.alpha = 0
+
+        getShareItem().uponQueue(.main) { shareItem in
```

Comment:
getShareItem() was previously in the ShareViewController and is now moved up the hierarchy as the InitialViewController needs to set the height based on the type of share data.